### PR TITLE
Use POST and body params to send cursor from browse

### LINKED ISSUE #525: 414 URI too long when browsing
When switching from the search API to the browse API (fetching > 1000 results), a request with a large number of search params can return a cursor (sample below) that is too long to be used in a valid URL.

The `browse(from cursor:String)` function returns a 414 HTTP status code. Is there a work around for this? 

Sample Cursor:
`AqY3ZmlsdGVycz0oc3RyYXRhJTNBJ2dyYWlsZWQnKSUyMEFORCUyMChkZXNpZ25lcnMuaWQlM0EnOTAxJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzYzMiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc4NDInJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNjUnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnOTk0JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzc5NSclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSczMDE3JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzU0JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzI2MSclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScyOTknJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTM2JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzMnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMjI2MyclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSczMDknJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTQ3JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzQ3JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzE4NzMnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMjI3JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzE1MCclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScxMDA5JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzUyNTInJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTE0JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzExNjknJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTA0MiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc0MDUnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTE3NyclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScxMzcyJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzMzMjAnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzE2NTEnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzE2MjgnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNDMxJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzUxJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzM5MDUnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzUnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzYnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNDU5MyclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScxMjkwOCclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc0MzgnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTI0MDUnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnODc0NyclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScyODYnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMjc3NiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScyNDIxJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzIyMzcnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNCclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScyMTk2JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzM0MTInJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNzknJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTgnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNTE5JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzU1MzEnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnOTYxMiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc1ODcnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMjA4MCclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSczMzknJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzIyJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzc1MDgnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNDY5JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzEnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnOTInJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzg4JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzE0NjAxJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzE0NzEnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzEyJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzEyNiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSczNTY2JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzE2MjAnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzQnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTQ2JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzIzMDgnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNjc0JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzE4MzkyJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzE4MjEnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMjA1JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzExMiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScxODUzOCclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc1MjI2JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzI5MzQnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzknJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzInJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNTgzJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzIxMzknJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMjk1JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzg4OSclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScxNjIxJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzE3ODU0JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzM5OSclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc2MTUnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTcxMyclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScxODcwMiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScxNTEwJyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzExOCclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScxMDAnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc4NSclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc3MTgnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNTE1JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzE0MjcnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMTMxNiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSczMzInJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMzA4ODQnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnOTkxMCclMjBPUiUyMGRlc2lnbmVycy5pZCUzQScxNiclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc2NjEnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnNTE3JyUyME9SJTIwZGVzaWduZXJzLmlkJTNBJzkwMCclMjBPUiUyMGRlc2lnbmVycy5pZCUzQSc1MzMnJTIwT1IlMjBkZXNpZ25lcnMuaWQlM0EnMjg4JyklMjBBTkQlMjAoY2F0ZWdvcnlfcGF0aF9zaXplJTNBJ2Zvb3R3ZWFyLmhpdG9wX3NuZWFrZXJzLjExLjUnJTIwT1IlMjBjYXRlZ29yeV9wYXRoX3NpemUlM0EnZm9vdHdlYXIuaGl0b3Bfc25lYWtlcnMuMTInJTIwT1IlMjBjYXRlZ29yeV9wYXRoX3NpemUlM0EnZm9vdHdlYXIuaGl0b3Bfc25lYWtlcnMuMTEnJTIwT1IlMjBjYXRlZ29yeV9wYXRoX3NpemUlM0EnZm9vdHdlYXIuZm9ybWFsX3Nob2VzLjExLjUnJTIwT1IlMjBjYXRlZ29yeV9wYXRoX3NpemUlM0EnZm9vdHdlYXIuZm9ybWFsX3Nob2VzLjEyJyUyME9SJTIwY2F0ZWdvcnlfcGF0aF9zaXplJTNBJ2Zvb3R3ZWFyLmZvcm1hbF9zaG9lcy4xMSclMjBPUiUyMGNhdGVnb3J5X3BhdGhfc2l6ZSUzQSdmb290d2Vhci5zbGlwX29ucy4xMS41JyUyME9SJTIwY2F0ZWdvcnlfcGF0aF9zaXplJTNBJ2Zvb3R3ZWFyLnNsaXBfb25zLjEyJyUyME9SJTIwY2F0ZWdvcnlfcGF0aF9zaXplJTNBJ2Zvb3R3ZWFyLnNsaXBfb25zLjExJyUyME9SJTIwY2F0ZWdvcnlfcGF0aF9za`

### PR DESCRIPTION
**Pull request by adamfraser**
Fixes #525

### CODE REVIEW

**Comment by robertmogos** [comment]
Can we write a test for this ?

File: Sources/AlgoliaSearch-Client/Index.swift [hunk 1]
```diff
@@ -535,10 +535,10 @@ import Foundation
     @objc(browseFromCursor:requestOptions:completionHandler:)
     @discardableResult public func browse(from cursor: String, requestOptions: RequestOptions? = nil, completionHandler: @escaping CompletionHandler) -> Operation {
         let path = "1/indexes/\(urlEncodedName)/browse"
-        let urlParameters = ["cursor": cursor]
-        return client.performHTTPQuery(path: path, urlParameters: urlParameters, method: .GET, body: nil, hostnames: client.readHosts, requestOptions: requestOptions, completionHandler: completionHandler)
+        let bodyParams = ["params": "cursor=\(cursor)"]
```
Comment (julienbourdeau) [inline Sources/AlgoliaSearch-Client/Index.swift]: You don't have to nest it under `params`, you can do:
```swift
let bodyParams = ["cursor": cursor]
```

Both will work so it's not so important, but we're trying to remove usage of `params: allTheThingsEncoded` to make the logs easier to read.

**Comment by julienbourdeau** [comment]
Hi @adamfraser,

Great catch! Thank you for your contribution. I have added a small comment, it's mostly for -future-selves when we read the logs in the dasboard. Otherwise the fix is the right approach 👍

**Comment by spinach** [comment]
moved to #594

**Pull request closed without merging**